Chronus/Assets/Scripts/LeverSwitch.cs

```diff
@@ -0,0 +1,98 @@
+
+public class LeverSwitch : MonoBehaviour
+{
+    public GameObject platform; 
+    public Transform lever;
+    public Transform baseObject;
+    public GameObject barrier;
+
+    private bool isActivated = false;
+    private Quaternion forwardRotation; 
+    private Quaternion backwardRotation; 
+    private bool playerAdjacent = false; 
+
+    private GameObject player; 
+    private Vector3 playerInitialTile; 
+
+    private void Start()
+    {
+        platform.SetActive(false);
+        barrier.SetActive(true); // Barrier is active initially to block entry
+
+        forwardRotation = Quaternion.Euler(130, 0, 0);
+        backwardRotation = Quaternion.Euler(50, 0, 0);
+
+        // Initial lever rotation
+        lever.rotation = backwardRotation;
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (other.CompareTag("Player"))
+        {
+            playerAdjacent = true;
+            player = other.gameObject;
+            playerInitialTile = player.transform.position; // Store player's position when adjacent
+        }
+    }
+
+    private void OnTriggerExit(Collider other)
+    {
+        if (other.CompareTag("Player"))
+        {
+            playerAdjacent = false;
+            player = null;
+        }
+    }
+
+    private void Update()
+    {
+        if (player != null && IsPlayerOnLeverTile())
+        {
+            player.transform.position = playerInitialTile; // Reset position to prevent entering the lever tile
+            return;
+        }
+
+        if (playerAdjacent && PlayerIsTryingToMoveTowardsLever())
+        {
+            playerInitialTile = player.transform.position;
+            ToggleLever();
+        }
+    }
+
+    private bool PlayerIsTryingToMoveTowardsLever()
+    {
+        if (player == null) return false;
+
+        // Calculate the direction from the player to the lever
+        Vector3 directionToLever = (lever.position - player.transform.position).normalized;
+
+        // Get the player's movement input direction
+        Vector3 playerInputDirection = Vector3.zero;
+
+        // Capture player input continuously
+        if (Input.GetKey(KeyCode.W)) playerInputDirection = Vector3.forward;
+        else if (Input.GetKey(KeyCode.S)) playerInputDirection = Vector3.back;
+        else if (Input.GetKey(KeyCode.A)) playerInputDirection = Vector3.left;
+        else if (Input.GetKey(KeyCode.D)) playerInputDirection = Vector3.right;
+
+        // Check if the player is pressing in the direction of the lever
+        return Vector3.Dot(directionToLever, playerInputDirection) > 0.5f;
+    }
+
+    private bool IsPlayerOnLeverTile()
+    {
+        return Vector3.Distance(player.transform.position, lever.position) < 0.5f;
+    }
+
+    private void ToggleLever()
+    {
+        isActivated = !isActivated;
+        lever.rotation = isActivated ? forwardRotation : backwardRotation;
+        platform.SetActive(isActivated);
+        barrier.SetActive(true);
+    }
+}
```

Comment:
레버 타일에 올라갈 수 있도록 처음에 코드를 짜서 조금 복잡해진 것 같은데,

지금 상황에서는
barrier, playerInitialTile, 그리고 reset position to prevent entering the lever tile 모두 필요가 없을 것 같습니다. 굳이 barrier로 넣고 setActive를 안 하고 그냥 LeverTile cube가 존재하기만 해도 잘 작동해서, "LeverTile cube"를 'lever'의 child로 넣고 그걸 prefab으로 새로 만들어주면 될 것 같아요

Reply:
수정했습니다